app/src/main/java/org/aplatanao/billing/cockpit/control/Log.java

```diff
@@ -1,15 +1,41 @@
 package org.aplatanao.billing.cockpit.control;
 
-import javafx.scene.control.TextArea;
+import javafx.collections.FXCollections;
+import javafx.collections.ObservableList;
+import javafx.scene.control.TableColumn;
+import javafx.scene.control.TableView;
+import javafx.scene.control.cell.PropertyValueFactory;
+import org.aplatanao.billing.cockpit.models.LogEntry;
 import org.springframework.stereotype.Component;
-
-import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.LocalDateTime;
 
 @Component
-public class Log extends TextArea {
+public class Log extends TableView {
+
+    private final ObservableList<LogEntry> data = FXCollections.observableArrayList();
+
+    public Log() {
+        TableColumn date = new TableColumn("Date");
+        date.setCellValueFactory(new PropertyValueFactory<LogEntry, String>("date"));
+        date.prefWidthProperty().bind(widthProperty().multiply(0.2));
+
+        TableColumn level = new TableColumn("Level");
+        level.setCellValueFactory(new PropertyValueFactory<LogEntry, String>("level"));
+        level.prefWidthProperty().bind(widthProperty().multiply(0.1));
+
+        TableColumn message = new TableColumn("Message");
+        message.setCellValueFactory(new PropertyValueFactory<LogEntry, String>("message"));
+        message.prefWidthProperty().bind(widthProperty().multiply(0.7));
+
+        getColumns().addAll(date, level, message);
+    }
 
     private void log(String level, String message) {
-        appendText(String.format("%s %s: %s\n", LocalDate.now(), level, message));
+        data.add(
+            new LogEntry(LocalDateTime.now().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss")), level, message)
+        );
+        setItems(data);
```

Comment:
data binding only requires a one-time assignment, so doing this once in the constructor is enough